coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1543,12 +1571,17 @@ func (s *server) notifyWorkspaceDeleted(ctx context.Context, workspace database.
 			slog.F("reason", reason), slog.F("workspace_id", workspace.ID), slog.F("build_id", build.ID))
 	}
 
+	labels := map[string]string{
+		"name":   workspace.Name,
+		"reason": reason,
+	}
+
+	if initiator != "" {
+		labels["initiator"] = initiator
+	}
```

Comment:
I don't understand why we need to do this; the template `{{ if .Labels.initiator }} ({{ .Labels.initiator }}){{end}}` I believe should not execute this branch if `initiator` is a zero value. Can you elaborate please?

If it's related to: `fix: do not publish initiator when autobuild deletes a workspace`, I think a blank `initiator` is not idiomatic because the initiator is the system.

Reply:
> I believe should not execute this branch if initiator is a zero value. Can you elaborate please?

I didn't want to keep empty fields in the payload. Similar behavior to "omit empty".

Reply:
> Is "system" already defined or can I just use it? I hope nothing breaks...

`initiator` is just a string; I don't believe anything uses it besides the email template.
`autobuild` (or whatever triggers this) might be a better initiator than `system`, come to think of it.